lib/ui/window/platform_configuration.h

```diff
@@ -633,6 +634,11 @@ class PlatformConfigurationNativeApi {
 
   static void RegisterBackgroundIsolate(int64_t root_isolate_token);
 
+  static Dart_Handle AddPlatformPortCallback(const std::string& name,
```

Comment:
SetChannelPort is a better name

Reply:
If the user clears their channel port (either by passing null here or by calling the function below)- is the behaviour that messages on that channel start to be delivered to the UI isolate again?

Reply:
What does this function return? Please document this.

Reply:
> If the user clears their channel port (either by passing null here or by calling the function below)- is the behaviour that messages on that channel start to be delivered to the UI isolate again?

Yes that's correct, this is also one of the parts of the implementation that gets a bit messy. I couldn't find a way for this to be cleared automatically when the registered background isolate finishes.